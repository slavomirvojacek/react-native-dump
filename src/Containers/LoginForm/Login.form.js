import React from 'react'
import { View, Button } from 'react-native'
import { reduxForm, Field } from 'redux-form'

import renderField from '../../FormUtils/field'
import { required } from '../../FormUtils/validation'

const LoginForm = ({pristine, submitting, handleSubmit, onSubmit, style}) => {
  return (
    <View style={style}>
      <Field name="email"
             component={renderField}
             type="email"
             label="Email"
             validate={[required]}/>
      <Field name="password"
             component={renderField}
             type="password"
             label="Password"
             validate={[required]}/>
      <Button title="Submit" onPress={handleSubmit(onSubmit)} disabled={pristine || submitting}/>
    </View>
  )
}

export { LoginForm }

export default reduxForm({
  form: 'Login', // destroyOnUnmount: true
})(LoginForm)
