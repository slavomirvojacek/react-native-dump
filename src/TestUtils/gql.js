const assertQueryOrMutation = (QoM) => {
  test('matches Query or Mutation', () => {
    expect(QoM).toMatchSnapshot()
  })
}

const assertGql = (Component) => {
  const { gql } = Component
  expect(gql instanceof Object).toEqual(true)
  Object
    .keys(gql)
    .forEach(k => assertQueryOrMutation(gql[k]))
}

export { assertGql }
