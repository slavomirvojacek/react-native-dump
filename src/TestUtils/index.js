export { assertGql } from './gql'
export { mockSetState } from './mocking'
export { assertSnapshots } from './snapshot'
