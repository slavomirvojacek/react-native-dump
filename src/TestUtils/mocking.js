// Sometimes, we might want to test internal state changes without having to render.
// This overrides the setState method to only update the state, without interacting with
// React's internals such as the scheduler.
const mockSetState = (Component) => {
  return class extends Component {
    setState (f) {
      if (typeof f === 'function') {
        this.state = {...this.state, ...f(this.state)}
      } else {
        this.state = {...this.state, ...f}
      }
    }
  }
}

export { mockSetState }
