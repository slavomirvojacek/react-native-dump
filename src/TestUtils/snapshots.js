import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

const assertSnapshots = (Component, props, desc, store = null) => {
  let getComponent = p => null

  if (store !== null) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

export { assertSnapshots }
