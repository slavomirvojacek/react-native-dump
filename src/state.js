import { createStore, applyMiddleware } from 'redux'
import createSagaMiddleware from 'redux-saga'

import reducers from './Redux/reducers'
import sagas from './Redux/sagas'

// create the saga middleware
const sagaMiddleware = createSagaMiddleware()
// mount it on the Store
const store = createStore(
  reducers,
  applyMiddleware(sagaMiddleware)
)

// then run the saga
sagaMiddleware.run(sagas)

export default store
