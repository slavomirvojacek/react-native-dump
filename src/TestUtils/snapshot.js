import React from 'react'
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer'

import { assertGql } from './gql'

import { isNull } from '../Utils/type'

const assertSnapshot = (Component, props, desc, store = null) => {
  let getComponent = p => null

  // In theory, if things are done correctly, this (wrapping
  // a component in Provider with a store) should never be needed.
  if (!isNull(store)) {
    getComponent = p => (
      <Provider store={store}>
        <Component {...p}/>
      </Provider>
    )
  } else {
    getComponent = p => <Component {...p}/>
  }

  test(desc, () => {
    const tree = renderer
      .create(getComponent(props))
      .toJSON()
    expect(tree).toMatchSnapshot()
  })
}

const assertSnapshots = (Component, configs, store = null) => {
  configs.forEach(config => assertSnapshot(Component, config.props, config.desc, store))
  if (!isNull(Component.gql)) assertGql(Component)
}

export { assertSnapshots }
