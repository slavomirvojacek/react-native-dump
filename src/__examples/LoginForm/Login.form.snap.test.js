import React from 'react'

import LoginForm from './Login.form'

import { assertSnapshots } from '../../TestUtils/snapshots'
import { formOnlyStore } from '../../state'

describe('LoginForm Snapshots', () => {
  const config = [
    {
      props: LoginForm.defaultProps,
      desc: 'renders correctly'
    },
    {
      props: {...LoginForm.defaultProps, pristine: true, submitting: false},
      desc: 'renders correctly when pristine and not submitting'
    },
    {
      props: {...LoginForm.defaultProps, pristine: false, submitting: true},
      desc: 'renders correctly when submitting and not pristine'
    },
    {
      props: {...LoginForm.defaultProps, pristine: false, submitting: false},
      desc: 'renders correctly when not pristine and not submitting'
    }
  ]

  config
    .forEach(conf => assertSnapshots(LoginForm, conf.props, conf.desc, formOnlyStore))
})
