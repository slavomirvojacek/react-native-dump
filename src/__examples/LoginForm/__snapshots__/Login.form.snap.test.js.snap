// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginForm Snapshots renders correctly 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Email
    </Text>
    <TextInput
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      style={
        Object {
          "height": 50,
        }
      }
      type="email"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Password
    </Text>
    <TextInput
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      style={
        Object {
          "height": 50,
        }
      }
      type="password"
      value=""
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`LoginForm Snapshots renders correctly when not pristine and not submitting 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Email
    </Text>
    <TextInput
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      style={
        Object {
          "height": 50,
        }
      }
      type="email"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Password
    </Text>
    <TextInput
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      style={
        Object {
          "height": 50,
        }
      }
      type="password"
      value=""
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`LoginForm Snapshots renders correctly when pristine and not submitting 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Email
    </Text>
    <TextInput
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      style={
        Object {
          "height": 50,
        }
      }
      type="email"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Password
    </Text>
    <TextInput
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      style={
        Object {
          "height": 50,
        }
      }
      type="password"
      value=""
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;

exports[`LoginForm Snapshots renders correctly when submitting and not pristine 1`] = `
<View
  style={undefined}
>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Email
    </Text>
    <TextInput
      name="email"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Email"
      style={
        Object {
          "height": 50,
        }
      }
      type="email"
      value=""
    />
  </View>
  <View
    style={
      Object {
        "marginBottom": 10,
        "minHeight": 50,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      disabled={false}
      ellipsizeMode="tail"
    >
      Password
    </Text>
    <TextInput
      name="password"
      onBlur={[Function]}
      onChange={[Function]}
      onDragStart={[Function]}
      onDrop={[Function]}
      onFocus={[Function]}
      placeholder="Password"
      style={
        Object {
          "height": 50,
        }
      }
      type="password"
      value=""
    />
  </View>
  <View
    accessibilityComponentType="button"
    accessibilityLabel={undefined}
    accessibilityTraits={
      Array [
        "button",
        "disabled",
      ]
    }
    accessible={true}
    hitSlop={undefined}
    isTVSelectable={true}
    nativeID={undefined}
    onLayout={undefined}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
    testID={undefined}
    tvParallaxProperties={undefined}
  >
    <View
      style={
        Array [
          Object {},
          Object {},
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        disabled={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#0C42FD",
              "fontSize": 18,
              "padding": 8,
              "textAlign": "center",
            },
            Object {
              "color": "#cdcdcd",
            },
          ]
        }
      >
        Submit
      </Text>
    </View>
  </View>
</View>
`;
